modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/UpgradeSCSSMediaQueryCheck.java

```diff
@@ -0,0 +1,114 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2023 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.source.formatter.check;
+
+import com.liferay.petra.string.StringPool;
+import com.liferay.petra.string.StringUtil;
+import com.liferay.source.formatter.check.util.JavaSourceUtil;
+
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * @author Michael Cavalcanti
+ */
+public class UpgradeSCSSMediaQueryCheck
+	extends BaseUpgradeMatcherReplacementCheck {
+
+	@Override
+	protected String formatMatcherIteration(
+		String content, String newContent, Matcher matcher) {
+
+		String mediaQuery = matcher.group(2);
+		String method = matcher.group(1);
+
+		if (mediaQuery.equals("respond-to")) {
+			return StringUtil.replace(
+				newContent, method, _replaceRespondToQuery(method));
+		}
+		else if (mediaQuery.equals("media-query")) {
+			return StringUtil.replace(
+				newContent, method, _replaceMediaQuery(method));
+		}
+
+		return StringUtil.replace(
+			newContent, method, _replaceGenericTypeQuery(method));
```

Comment:
hi @michaelmdelcavalcanti, please rename to `_replaceGenericTypeMixin` or `_replaceGenericType`